flytekit/common/tasks/sagemaker/hpo_job_task.py

```diff
@@ -21,13 +23,13 @@
 
 class SdkSimpleHyperparameterTuningJobTask(_sdk_task.SdkTask):
     def __init__(
-        self,
-        max_number_of_training_jobs: int,
-        max_parallel_training_jobs: int,
-        training_job: SdkBuiltinAlgorithmTrainingJobTask,
-        retries: int = 0,
-        cacheable: bool = False,
-        cache_version: str = "",
+            self,
+            max_number_of_training_jobs: int,
+            max_parallel_training_jobs: int,
+            training_job: typing.Union[SdkBuiltinAlgorithmTrainingJobTask, CustomTrainingJobTask],
```

Comment:
This is really smart! Could you please add a small unit test for that if you haven’t done that already?